File: code.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
int main()
{

    // Initialiser la graine du générateur aléatoire
    srand(time(0));

    // Générer un nombre aléatoire entre 0 et 100
    int nombre_secret = rand() % 101;

    // Boucle pour demander à l'utilisateur de deviner le nombre
    while (1) {
        //printf(" %d ",nombre_secret);//pour le test
        int reponse;
        printf("Devinez le nombre (entre 0 et 100) : ");
        scanf("%d", &reponse);

        // Vérifier si la réponse est correcte et donner des indications
        if (reponse < nombre_secret) {
            printf("plus grand !\n");
        } else if (reponse > nombre_secret) {
            printf("plus petit \n");
        } else {
            printf("Bravo \n");
            break;  // Sortir de la boucle si la réponse est correcte
        }
    }

    return 0;
}

File: code.java
package code;
import java.util.Random;
import java.util.Scanner;

public class code {

	public static void main(String[] args) {
		
		Random rand = new Random();
		 int nombreAleatoire = rand.nextInt(101); // génère un nombre aléatoire entre 0 et 100
	     System.out.println(nombreAleatoire);//j'ai reservé cette ligne pour les test
	     System.out.println("entrez un nnombre :");
	     Scanner cl = new Scanner(System.in);
	     int nombre = cl.nextInt();
	     if(nombreAleatoire > nombre)
	     {
	    	 System.out.println("plus grand");
	     }
	     else if(nombreAleatoire < nombre ) {
	    	 System.out.println("plus petit");
	     }
	     else if(nombreAleatoire == nombre )
	     {
	    	 System.out.println("Bravo!");
	     }
	}

}
